roles/sys/repositories/tasks/os/SUSE/repositories.yml

```diff
@@ -42,6 +42,25 @@
     repo: "{{ repos[r] }}"
   ignore_errors: true
 
+- name: Find existing EDB repositories
+  find:
+    paths: /etc/zypp/repos.d/
+    patterns: 'enterprisedb-*.repo'
+    file_type: "file"
+  register: existing_edb_repos
+
+- name: remove unwanted EDB repositories
+  file:
+    name: "{{ item.path }}"
+    state: absent
+  loop: "{{ existing_edb_repos.files }}"
+  loop_control:
+    label: "{{ item.path | basename }}"
+  vars:
+    repo: "{{ item.path|basename|split('.')|first|split('-') }}"
+  when: >
+    repo[1] not in edb_repositories
```

Comment:
```suggestion
    repo|last not in edb_repositories
```

Reply:
Last won't work since it will take arch part of the filename instead of repo name
basename: `enterprisedb-enterprise-noarch.repo`
split('.')|first: `enterprisedb-enterprise-noarch`
repo|last: `noarch`
it will always remove all existing repo and add the defined one right after.